tests/beaker_tests/cisco_portchannel_global/test_portchannel_global.rb

```diff
@@ -12,509 +12,323 @@
   },
 }
 
-#################################################################
-# HELPER FUNCTIONS
-#################################################################
-
-# Full command string for puppet resource command
-def puppet_resource_cmd
-  PUPPET_BINPATH + 'resource cisco_portchannel_global'
-end
-
-def build_manifest_portchannel_global(tests, id)
-  manifest = tests[id][:manifest_props]
-  tests[id][:resource] = tests[id][:resource_props]
-
-  tests[id][:title_pattern] = id if tests[id][:title_pattern].nil?
-  logger.debug(
-    "build_manifest_portchannel_global :: title_pattern:\n" +
-             tests[id][:title_pattern])
-  tests[id][:manifest] = "cat <<EOF >#{PUPPETMASTER_MANIFESTPATH}
-  node 'default' {
-    cisco_portchannel_global { 'default':
-      #{manifest}
-    }
-  }
-EOF"
+tests[:non_default][:manifest_props].merge!(manifest_non[:n3k]) if platform[/n3k/]
+tests[:non_default][:manifest_props].merge!(manifest_non[:n56k]) if platform[/n(5|6)k/]
+tests[:non_default][:manifest_props].merge!(manifest_non[:n8k]) if platform[/n8k/]
+tests[:non_default][:manifest_props].merge!(manifest_non[:n9k]) if platform[/n9k/]
+
+def unsupported_properties(tests, _id)
+  unprops = []
+  if platform[/n7k/]
+    unprops <<
+      :concatenation <<
+      :hash_poly <<
+      :resilient <<
+      :symmetry
+  elsif platform[/n(5|6)k/]
+    unprops <<
+      :asymmetric <<
+      :concatenation <<
+      :hash_distribution <<
+      :load_defer <<
+      :resilient <<
+      :rotate <<
+      :symmetry
+  elsif platform[/n3k/]
+    unprops <<
+      :asymmetric <<
+      :concatenation <<
+      :hash_distribution <<
+      :hash_poly <<
+      :load_defer <<
+      :rotate
+  elsif platform[/n8k/]
+    unprops <<
+      :asymmetric <<
+      :concatenation <<
+      :hash_distribution <<
+      :hash_poly <<
+      :load_defer <<
+      :resilient <<
+      :symmetry
+  elsif platform[/n9k/]
+    unprops <<
+      :asymmetric <<
+      :hash_distribution <<
+      :hash_poly <<
+      :load_defer
+  end
+  unprops << :resilient << :symmetry if
+    platform[/n3k/] && tests[:resilient_unsupported]
+  unprops
 end
 
-def test_harness_portchannel_global(tests, id)
-  tests[id][:resource_cmd] = puppet_resource_cmd
-
-  # Build the manifest for this test
-  build_manifest_portchannel_global(tests, id)
-
-  # test_harness_common(tests, id)
-  test_manifest(tests, id)
-  test_resource(tests, id)
-  test_idempotence(tests, id)
-
-  tests[id][:ensure] = nil
+if platform[/n3k/]
+  tests[:resilient_unsupported] =
+    resource_probe(agent,
+                   'cisco_portchannel_global default resilient=true',
+                   'ERROR: This feature is not supported on this platform.')
 end
 
 #################################################################
 # TEST CASE EXECUTION
 #################################################################
-test_name "TestCase :: #{testheader}" do
+test_name "TestCase :: #{tests[:resource_name]}" do
```

Comment:
You should have a teardown and cleanup at the beginning here, e.g. https://github.com/cisco/cisco-network-puppet-module/blob/develop/tests/beaker_tests/cisco_vlan/test_vlan.rb#L127

This is a new format we recently added. The teardown will run even if the test hits an error. The initial cleanup is optional: you can either do it here or rely on a `:preclean` entry in the testcase. Either way you should have some kind of cleanup before & after this test runs.